integration-tests/tests/src/tests/enums.ts

```diff
@@ -0,0 +1,73 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import {
+  ConnectionState,
+  NumericLogLevel,
+  OpenRealmBehaviorType,
+  OpenRealmTimeOutBehavior,
+  SessionStopPolicy,
+} from "realm";
+
+describe("Enums", function () {
+  describe("ConnectionState", function () {
+    it("is accessible", function () {
+      expect(typeof ConnectionState === "object");
+      expect(typeof ConnectionState.Disconnected === "string");
```

Comment:
Could be redundant to check all enum values, feel free to add thoughts about this.

Reply:
I'd vote for keeping them (i.e. testing all values). Even though the issue mentioned asserting that the value is a string, we'd get more accurate tests if we compare it to the actual string value (e.g. `disconnected`).

On a smaller note, perhaps we want to structure them more toward "natural language"? E.g.:
```ts
// This will pass for e.g. arrays as well.
expect(typeof ConnectionState).to.equal("object");

// This will not pass for arrays.
expect(ConnectionState).to.be.an("object");

expect(ConnectionState.Disconnected).to.equal("disconnected");
expect(ConnectionState.Connecting).to.equal("connecting");
// Fixed typo from `Connecting` to `Connected`
expect(ConnectionState.Connected).to.equal("connected");
```

Reply:
yeah went through this a bit too fast 🤦  using mocha as extensively as possible is what we've already done in `test-migration-master` so should aim for that! Thanks for pointing it out

Reply:
Np 👍  (perhaps `to.deep.equal()` could test all those cases at once as well)

Reply:
> Could be redundant to check all enum values.

I'm probably feeling this a bit. If these tests fail it'll either be because we (hopefully consciously) changed the enums or the TS compiler is broken (in which case this will be the least of our worries). These tests are in risk of becoming outdated if we add new values to the enum.

The main reason we had these issues on `master` was because we declared types separately from implementation (in JS), so I probably don't feel the need for this to be a part of our test suite. But then again - they are written now, so they don't hurt too much to keep 😅